tests/components/panasonic_viera/test_init.py

```diff
@@ -49,32 +51,46 @@ def get_mock_remote(device_info=MOCK_DEVICE_INFO):
     async def async_create_remote_control(during_setup=False):
         return
 
-    mock_remote.async_create_remote_control = async_create_remote_control
+    mock_remote.async_create_remote_control = AsyncMock(
+        side_effect=async_create_remote_control
+    )
 
     async def async_get_device_info():
         return device_info
 
-    mock_remote.async_get_device_info = async_get_device_info
+    mock_remote.async_get_device_info = AsyncMock(side_effect=async_get_device_info)
 
     async def async_turn_on():
         return
 
-    mock_remote.async_turn_on = async_turn_on
+    mock_remote.async_turn_on = AsyncMock(side_effect=async_turn_on)
 
     async def async_turn_off():
         return
 
-    mock_remote.async_turn_on = async_turn_off
+    mock_remote.async_turn_on = AsyncMock(side_effect=async_turn_off)
 
     async def async_send_key(key):
         return
 
-    mock_remote.async_send_key = async_send_key
+    mock_remote.async_send_key = AsyncMock(side_effect=async_send_key)
 
     return mock_remote
 
 
-async def test_setup_entry_encrypted(hass):
+@pytest.fixture(name="mock_remote")
+def mock_remote_fixture():
```

Comment:
If we move the fixture to a module `conftest.py` in this package we don't need to import the fixture. It will be available for use in all submodules of the test package by default.

Reply:
hey there! just did that, as well as some extra organization (for example, I standardised most of the configuration dictionaries)

it's not perfect though, as some occurrences aren't yet being checked (for example, `turn_on_action` is always compared to `None`), but I guess that should be in a different pull request.

thank you for all your help, by the way! 😌